docs/tech-preview-blog-post.md

```diff
@@ -0,0 +1,130 @@
+that data needs to be formatted. 
+In many languages, including English, the message itself needs to be altered
+to make it grammatically correct.
+
+For example, if a message in English might read:
+
+> Your item had **1,023** views on **April 8, 2024**.
+
+The equivalent message in French might read:
+
+> Votre article a eu **1 023** vues le **8 avril 2024**.
+
+Or Japanese:
+
+> あなたのアイテムは **2024 年 4 月 8 日**に **1,023** 回閲覧されました。
+
+But even in English, there are grammatical variations required:
+
+> Your item had _no views_...
+> 
+> Your item had 1 _view_...
+> 
+> Your item had 1,043 _views_...
+
+Once messages have been created, they need to be translated into the various
+languages and adapted for the various cultures around the world.
+Previously there was not widely adopted standard
+and existing formats provided only rudimentary support for managing
+the variations needed by other languages,
+it could be difficult for translators to do their work effectively.
+
+For example, the same message shown above, needs a different set of variations
+in order to support Polish:
+
+> Twój przedmiot nie _ma_ żadnych _wyświetleń_.
+> 
+> Twój przedmiot _miał_ 1 _wyświetlenie_.
+> 
+> Twój przedmiot _miał_ 2 _wyświetlenia_.
+> 
+> Twój przedmiot _ma_ 5 _wyświetleń_.
+
+
+MessageFormat 2 makes it easy to write messages like this
+without developering needing to know about such language variation.
+In fact, developers don't need to learn about any of the language
+and formatting variations needed by languages other than their own
+nor write code that manipulates formatting.
+
+MessageFormat 2 messages can be simple strings:
+```
+    Hello, world!
+```
+
+A message can also include _placeholders_ that are replaced by user-provided values:
+```
+    Hello {$user}!
+```
+
+The user-provided values can be transformed or formatted using functions:
+```
+    Today is {$date :datetime}
+    Today is {$date :datetime weekday=long}.
+```
+
+Messages can use a function (called a _selector_) to choose between different
+different versions of a message.
+These allow messages to be tailored to the grammatical (or other) requirements of 
+a given language:
+```
+    .match {$count :integer}
+    0   {{You have no views.}}
+    one {{You have {$count} view.}}
+    *   {{You have {$count} views.}}
+```
+
+Unlike previous versions of MessageFormat, MessageFormat 2 is designed for
+extension by implementers and even end users.
+This means that new functionality can be added to messages without modifying
+either existing messages or, in some cases, even the core library containing the 
+MessageFormat 2 code.
+
+MessageFormat 2 provides a rich and extensible set of functionality
+to permit the creation of natural-sounding, grammatically-correct, 
+messages, while enabling rapid, accurate translation
+and extension using new and improved internationalization functionality
+in any computing system.
```

Comment:
This paragraph feels like it could be lifted up much higher, as it's a pretty good summary of MF2.

Reply:
I suggest moving it up just one paragraph. It will only make sense to a lot of people once they have read the intro, which motivates it.

```suggestion
```